File: src/main/java/com/mbayou/kick4k/events/type/ReplyInfo.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ReplyInfo {
    private final String messageId;
    private final String content;
    private final EventUser sender;

    @JsonCreator
    public ReplyInfo(@JsonProperty("message_id") String messageId,
                     @JsonProperty("content") String content,
                     @JsonProperty("sender") EventUser sender) {
        this.messageId = messageId;
        this.content = content;
        this.sender = sender;
    }

    public String getMessageId() {
        return messageId;
    }

    public String getContent() {
        return content;
    }

    public EventUser getSender() {
        return sender;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/ReplyInfo.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ReplyInfo @JsonCreator constructor(
    @JsonProperty("message_id") val messageId: String?,
    @JsonProperty("content") val content: String?,
    @JsonProperty("sender") val sender: EventUser?,
)
